apps/aecore/lib/aecore/chain/block.ex

```diff
@@ -32,4 +33,62 @@ defmodule Aecore.Chain.Block do
     header = genesis_header()
     %Block{header: header, txs: []}
   end
+
+  @spec rlp_encode(Block.t()) :: binary() | {:error, String.t()}
+  def rlp_encode(%Block{} = block) do
+    header_bin = Header.header_to_binary(block.header)
+
+    txs =
+      for tx <- block.txs do
+        SignedTx.rlp_encode(tx)
+      end
+
+    [
+      type_to_tag(Block),
+      block.header.version,
+      header_bin,
+      txs
+    ]
+    |> ExRLP.encode()
+  end
+
+  def rlp_encode(_) do
+    {:error, "Invalid block or header struct"}
+  end
+
+  @spec rlp_decode(binary()) :: Block.t() | {:error, String.t()}
+  def rlp_decode(values) when is_binary(values) do
+    [tag_bin, ver_bin | rest_data] = ExRLP.decode(values)
```

Comment:
What if the result list is too short? We should probably handle that gracefully.